index.d.ts

```diff
@@ -31,17 +31,17 @@ declare namespace megajs {
         inbox: MutableFile;
         mounts: MutableFile[];
         files: { [id in string]: MutableFile };
-        RSAPrivateKey: (number | number[])[];        
+        RSAPrivateKey: (number | number[])[];
         constructor(options: StorageOpts, cb?: errorCb);
         toJSON(): StorageJSON;
-        close(cb?: noop): void;
+        close(cb?: noop): Promise<void>;
         static fromJSON(json: StorageJSON): Storage;
         mkdir(opt: mkdirOpts | string, cb?: errorCb): void;
         upload(opt: uploadOpts | string, buffer?: BufferString, cb?: uploadCb): Writable;
-        login(cb: (error: err, storage: this) => void): void;
-        getAccountInfo(cb: (error: err, account: accountInfo) => void): accountInfo;
-        reload(cb: (error: err, mount: MutableFile[]) => void): void | this;
-        reload(force: boolean, cb: (error: err, mount: MutableFile[]) => void): void | this;
+        login(cb: (error: err, storage: this) => void): Promise<this>;
+        reload(cb: (error: err, mount: MutableFile[]) => void): Promise<MutableFile[]>;
+        getAccountInfo(cb: (error: err, account: accountInfo) => void): Promise<accountInfo>;
```

Comment:
Because those methods return promises now their callbacks are optional, otherwise you will need to set a callback when using promises, which would be weird.